src/utils.cpp

```diff
@@ -924,6 +924,20 @@ std::string utils::getcwd()
 	return RustString(rs_getcwd());
 }
 
+bool utils::read_text_file(const std::string& filename,
+	std::vector<std::string>& contents, std::string& error_message)
```

Comment:
Could this return `nonstd::expected<std::vector<std::string>, std::string>` instead? That'd be a bit more foolproof I think.

Reply:
I somehow didn't notice that we have `nonstd::expected`.
Good idea, done in https://github.com/newsboat/newsboat/pull/1332/commits/fdb58e07c90d8e8addb14835c0c337f78f707faf.